### PR TITLE
Bugfix/enable teams message failed builds

### PR DESCRIPTION
**Pull request by erbenjak**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,10 +46,10 @@ After preparation of your local environment the script will execute the jscm com
 
 It will execute both possible command:
 
-- teams-develop (in failed/successful state)
-- teams-production (in failed/successful state)
+- teams-develop (in failed/successful state/without artifacts)
+- teams-production (in failed/successful state/without artifacts)
 
-&rarr; This will result in 4 messages being sent to the specified teams channel
+&rarr; This will result in 4(2 failed/2 successful) messages being sent to the specified teams channel as well as 2 exceptions exceptions.
```
Comment (jkoenig134) [inline README.md:52]: exceptions exceptions?

File: src/index.ts [hunk 2]
```diff
@@ -80,14 +83,23 @@ async function run() {
       () => {
         return;
       },
-      async (args) =>
+      async (args) => {
+        if (failedState) {
+          console.log("Cannot send production message for a build, which did not produce an artifact.");
+          process.exit(1);
```
Comment (jkoenig134) [inline src/index.ts:89]: thats not right. we don't have artifacts in prod at all. But we want to send a message for failed prod builds.

File: src/index.ts [hunk 3]
```diff
@@ -80,14 +83,23 @@ async function run() {
...
+        }
+        if (!buildWasSuccessful) {
+          console.log("Cannot send production message for failed build.");
+          process.exit(1);
```
Comment (jkoenig134) [inline src/index.ts:93]: why? :)

File: src/index.ts (hunk 2 above)
Comment (erbenjak) [reply src/index.ts:89]: we still have artifacts in the sence that a build is created on codemagic and the corresponding file is passed on. This indicates whether a successful build did occur. So far we do not send messages for unsuccesfull production builds. However I'm not entirely sure as to why. I guess we argued that this pipeline should only be triggered once the dev-pipeline ran succesfully, therefore it should be impossible to reach this case.

Comment (erbenjak) [reply src/index.ts:89]: @jkoenig134 should we simply add a negative Production message?

Comment (jkoenig134) [reply src/index.ts:89]: I looked in the code. we already have one.

Comment (erbenjak) [reply src/index.ts:89]: I dont think we have. We have a negative build message - however a message about a failed prodction should be different from this.

When we call "jscm teams-production" - the message is produced by the following snippet:
image

Comment (jkoenig134) [reply src/index.ts:89]: ok, in that case we defenitely need that message.

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**